eth/filters/api.go

```diff
@@ -330,18 +330,14 @@ func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([
 		filter = NewBlockFilter(api.backend, *crit.BlockHash, crit.Addresses, crit.Topics)
 	} else {
 		// Convert the RPC block numbers into internal representations
-		var (
-			begin int64
-			end   int64
-		)
 		if crit.FromBlock == nil {
-			begin = int64(rpc.LatestBlockNumber)
+			crit.FromBlock = big.NewInt(rpc.LatestBlockNumber.Int64())
 		}
 		if crit.ToBlock == nil {
-			end = int64(rpc.LatestBlockNumber)
+			crit.ToBlock = big.NewInt(rpc.LatestBlockNumber.Int64())
 		}
```

Comment:
Please keep the variables and update those instead of the criteria. I don't think in general it's a good idea to modify input parameters:

```go
 		begin := rpc.LatestBlockNumber.Int64()
 		if crit.FromBlock != nil {
 			begin = crit.FromBlock.Int64()
 		}
 		end := rpc.LatestBlockNumber.Int64()
 		if crit.ToBlock != nil {
 			end = crit.ToBlock.Int64()
 		}
```